components/reddit/common.js

```diff
@@ -0,0 +1,32 @@
+const reddit = require("./reddit.app");
+const get = require("lodash.get");
+
+module.exports = {
+  dedupe: "unique",
+  props: {
+    reddit,
+    db: "$.service.db",
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for events on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // by default, run every 15 minutes.
+      },
+    },
+  },
+  methods: {
+    async emitRedditEvent(reddit_event) {
+      const metadata = this.generateEventMetadata(reddit_event);
+      this.$emit(reddit_event, metadata);
+    },
+    wereThingsPulled(reddit_things) {
+      const things = get(reddit_things, "data.children");
+      return things && things.length;
+    },
+    did4xxErrorOccur(err) {
+      const status = get(err, "response.status");
+      return status && status >= 400;
+    },
```

Comment:
If `status === 500`, then this will still be `true`.

Also as a general rule, we tend to deal with certain status codes within the methods that make the actual call (`_makeRequest` in your case), and let the exception bubble up in all other cases. Statuses like `429` are an example, where we don't necessarily want to expose or abort the execution of a component when a simple retry should solve it.